eogrow/cli.py

```diff
@@ -138,27 +130,38 @@ def ray(
         cli_variables: Tuple[str],
         test_patches: Tuple[int],
     ) -> None:
-        """Command for running an eo-grow pipeline on a remote Ray cluster of AWS EC2 instances
+        """Command for running an eo-grow pipeline on a remote Ray cluster of AWS EC2 instances. The provided config is
+        fully constructed and uploaded to the cluster head in the `~/.synced_configs/` directory, where it is then
+        executed. A custom suffix is added to distinguish runs which use the same config multiple times.
 
         \b
         Example:
             eogrow-ray cluster.yaml config_files/config.json
         """
+        if start_cluster:
+            subprocess.check_call(f"ray up -y {cluster_yaml}", shell=True)
```

Comment:
what happens if the cluster is up? will this reset it? it's been a while since i worked with ray

Reply:
Hmm, I though the same rules apply as before with `ray exec` but [the docs](https://docs.ray.io/en/latest/cluster/cli.html#ray-exec-doc) state that this is meant as an on-need utility, but I couldn't find the exact implementation of what it does under the hood.

`ray up` can do some restarts, which in my experience is not desired and becomes complicated if docker is used, even if `--no-restart` is provided.

perhaps it would be safer to remove this option, or run the lines above only in case the cluster is not running (checking with `ray get_head_ip`)

Reply:
if there is an easy way to check if the cluster is online, then let's do that
otherwise let's remove --start and --stop flags IMO

Reply:
Managed to define some ray utility functions for this so that it looks OK in cli. I tested it out and it works as expected:
- managed to get both True and False for the `is_cluster_running`
- `eogrow-ray` spinning up a cluster if `--start` is provided and no cluster is running
  - if `--start` is not provided, the process fails with an appropriate error message

Check the last commit https://github.com/sentinel-hub/eo-grow/pull/218/commits/433e9d22604d1975cb37476a0d9585f5f578f640